custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -0,0 +1,324 @@
+        rows = []
+        if self.config['location_id']:
+            supply_points = SQLLocation.objects.filter(parent__location_id=self.config['location_id'])\
+                .order_by('name').exclude(supply_point_id__isnull=True)
+            for sp in supply_points:
+                if sp.location_type == 'facility':
+                    cls = StockLevelsReport
+                else:
+                    cls = StockStatus
+                url = make_url(
+                    cls,
+                    self.config['domain'],
+                    '?location_id=%s&filter_by_program=%s&startdate=%s'
+                    '&enddate=%s&report_type=%s&filter_by_product=%s',
+                    (sp.location_id, self.config['program'], self.config['startdate'],
+                    self.config['enddate'], self.config['report_type'],
+                    '&filter_by_product='.join(self.config['products'])))
+
+                row = [link_format(sp.name, url)]
+                for p in products:
+                    stock = StockState.objects.filter(sql_product=p, case_id=sp.supply_point_id)\
+                        .order_by('-last_modified_date')
+
+                    if stock:
+                        if stock[0].get_monthly_consumption():
+                            row.append(int(stock[0].stock_on_hand / stock[0].get_monthly_consumption()))
+                        else:
+                            row.append('-')
+                    else:
+                        row.append(0)
+                rows.append(row)
+        return rows
+
+
+class StockoutsProduct(EWSData):
+
+    slug = 'stockouts_product'
+    title = 'Stockout by Product'
+    show_chart = True
+    show_table = False
+    chart_x_label = 'Months'
+    chart_y_label = 'Facility count'
+
+    @property
+    def headers(self):
+        return []
+
+    @property
+    def rows(self):
+        rows = {}
+        if self.config['location_id']:
+            supply_points = get_supply_points(self.config['location_id'], self.config['domain'])
+            products = get_products(self.config['program'],
+                                    self.config['products'],
+                                    self.config['domain']).order_by('code')
+
+            for product in products:
+                rows[product.code] = []
+
+            for d in get_second_week(self.config['startdate'], self.config['enddate']):
+                for product in products:
+                    st = StockTransaction.objects.filter(case_id__in=supply_points,
+                                                         sql_product=product,
+                                                         report__date__range=[d['start_date'],
+                                                                              d['end_date']],
+                                                         type='stockonhand',
+                                                         stock_on_hand=0).count()
```

Comment:
@twymer 

I want count the facilities which are stockout in second week in every month from data range.